### ISSUE TITLE
Constant SIGSEGV libjvm.so

### ISSUE THREAD

**Issue by TomLewis** [AdoptOpenJDK/openjdk-support#143]
## Summary

I have used the same stack for years and in the last few months been getting constant random JVM crashes with my minecraft server, ive run it for 7 years and been running trial and error for ages, but im at my wits end with figuring this out as there is next to no information.
I have found that when the JVM crashes if its got a YAML config in memory from plugins that have worked for years, it can corrupt the YAML file when we next boot back up.

```
2020-07-15 19:32:09 | [RAW] #
2020-07-15 19:32:09 # server jvm crashed, restarting...
2020-07-15 19:32:09 # stopping survival (caused by jvm crash)
2020-07-15 19:32:09 | [RAW] # A fatal error has been detected by the Java Runtime Environment:
2020-07-15 19:32:09 | [RAW] #
2020-07-15 19:32:09 | [RAW] #  SIGSEGV (0xb) at pc=0x00007fb4ac9905ab, pid=25188, tid=25190
2020-07-15 19:32:09 | [RAW] #
2020-07-15 19:32:09 | [RAW] # JRE version: OpenJDK Runtime Environment (11.0.7+10) (build 11.0.7+10-post-Debian-3deb10u1)
2020-07-15 19:32:09 | [RAW] # Java VM: OpenJDK 64-Bit Server VM (11.0.7+10-post-Debian-3deb10u1, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
2020-07-15 19:32:09 | [RAW] # Problematic frame:
2020-07-15 19:32:09 | [RAW] # V  [libjvm.so+0x7325ab]
2020-07-15 19:32:09 | [RAW] #
2020-07-15 19:32:09 | [RAW] # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
2020-07-15 19:32:09 | [RAW] #
2020-07-15 19:32:09 | [RAW] # An error report file with more information is saved as:
2020-07-15 19:32:09 | [RAW] # /home/username/folder/hs_err_pid25188.log
2020-07-15 19:32:09 | [RAW] [thread 25216 also had an error]
2020-07-15 19:32:09 | [RAW] #
2020-07-15 19:32:09 | [RAW] # If you would like to submit a bug report, please visit:
2020-07-15 19:32:09 | [RAW] #   https://bugs.debian.org/openjdk-11
2020-07-15 19:32:09 | [RAW] #
```
The only consistency is a SIGSEGV with libjvm.so, I am no Java developer so I have no idea what either means.

Here are a few copies of my hs_err_pid logs that have been created today, I have had 7 crashes today and I had 19 yesterday.

- https://gist.github.com/FrozenBeard/98b0e8d536ead178c16bba0a7bfc72a7
- https://gist.github.com/FrozenBeard/cee5d9911101e1da47c535c67bf7766b
- https://gist.github.com/FrozenBeard/ee2ebf905d0dfce0134650ea00d6757a

## Steps to reproduce

Run a paper 1.12.2 Minecraft server https://papermc.io/legacy 
Using aikar's flags. https://aikar.co/2018/07/02/tuning-the-jvm-g1gc-garbage-collector-flags-for-minecraft/

Copy of flags I use;
```
java -Xms10G -Xmx10G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -jar paperclip.jar nogui
```

## Expected results
Not crash multiple times a day.

## Actual results
Crashing multiple times a day, sometimes 10-15 times day.

Java version:

```
java -version
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Debian-3deb10u1)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Debian-3deb10u1, mixed mode, sharing)
```

Your operating system and platform:
```
Host: Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz, 8 cores, 62G, Debian GNU/Linux 10 (buster)
```
Did it work before:

Yeah, I have been using 1.12.2 Paper Since September 2017, smoothly with no JVM crashes, Its near impossible for a plugin to create a JVM crash, if a plugin crashes the server it does not crash the JVM it just crashes the server with its own error logs, its very rare to get a JVM crash, which is why im so worried as im now getting 10+ a day on some days, randomly with no consistently to it.

Did you test with other Java versions:

No, not yet, which versions should I test with? how Do I do this?

**Comment by aahlenst** [comment]
@FrozenBeard Thanks for taking the time to fill out the entire form! So, one thing I see is that you are using a build created by the Debian package maintainers. If so, we cannot help that much because it's not from us. What we could try is to find out whether the problem is build or version specific and then report to the right folks. Could you download OpenJDK from https://adoptopenjdk.net/upstream.html, expand it and run your Minecraft server with that one and see if it continues to crash?

**Comment by FrozenBeard** [comment]
Hey @aahlenst its been so long since I setup java now! I checked and I setup using adoptopenjdk using the sources.list in debian so I could get updates through apt update, this is whats in my sources list at present, is there a different repo url for upstream?

```deb https://adoptopenjdk.jfrog.io/adoptopenjdk/deb/ buster main```

**Comment by aahlenst** [comment]
`apt install adoptopenjdk-11-hotspot` is correct for using AdoptOpenJDK. To change versions, use `update-java-alternatives`. It lists all available versions and allows changing them globally. With `update-alternatives` alone things are more complicated.

We do not distribute the upstream versions as Debian package. If you expand the downloaded tarball, you can run `folder/bin/java <yourarguments>`. This is the recommended approach for testing.

**Comment by tmancill** [comment]
@FrozenBeard Did switching to `adoptopenjdk-11-hotspot` fix the crashes?

If so, I will take this and open a bug report against Debian's openjdk-11-jre package.  Thank you for providing detailed information.  

In all 3 crashes you provided, the active thread is GCTaskThread.  If you are still seeing crashes with the AdoptOpenJDK build, I wonder if you might have a hardware problem.

**Comment by FrozenBeard** [comment]
> 
> 
> @FrozenBeard Did switching to `adoptopenjdk-11-hotspot` fix the crashes?
> 
> If so, I will take this and open a bug report against Debian's openjdk-11-jre package. Thank you for providing detailed information.
> 
> In all 3 crashes you provided, the active thread is GCTaskThread. If you are still seeing crashes with the AdoptOpenJDK build, I wonder if you might have a hardware problem.

One of two things fixed it, I both switched to adoptopenjdk and I removed all my Java flags, so one of the things fixed it. 
Since performance hasn't really taken a hit I've not bothered to put the flags back on. I don't see the point now, so if anyone else finds this, do both!

But either way it was solved and I haven't had a crash since! whats better is now when any plugin crashses my server, I get proper crashes again and not bloody JVM crashes! so im over the moon.

**Comment by tmancill** [comment]
> One of two things fixed it, I both switched to adoptopenjdk and I removed all my Java flags, so one of the things fixed it.
> Since performance hasn't really taken a hit I've not bothered to put the flags back on. I don't see the point now, so if anyone else finds this, do both!
> 
> But either way it was solved and I haven't had a crash since! whats better is now when any plugin crashses my server, I get proper crashes again and not bloody JVM crashes! so im over the moon.

@FrozenBeard Thank you for confirming. Once I'm able to reproduce the issue, I will file a bug against Debian.

**Comment by TomLewis** [comment]
Ahoy, not sure if I should open a new ticket or not, but out of this blue it started happening again, this time I am using AdoptOpenJDK (11.0.8+10) (build 11.0.8+10).

I got 6 crashes all next to each other and have no idea how to diagnose whats causing it.

https://gist.github.com/TomLewis/9429f79ad2439cab0fd9da174a5d26f7
https://gist.github.com/TomLewis/c4207627b26a47584d3de00c60ca5bcf
https://gist.github.com/TomLewis/938c0654740bac27a365556f65d5229e
https://gist.github.com/TomLewis/6d6a3f42e1941af3c208fad63b765dae
https://gist.github.com/TomLewis/26ca096f393b6b0f369c3e31793d1a2f
https://gist.github.com/TomLewis/cbb358ed834f9638035b562d2694e3c2

**Comment by karianna** [comment]
Hi @TomLewis - can you try with a https://adoptopenjdk.net/nightly.html build of 11.0.9 EA?

**Comment by TomLewis** [comment]
@karianna Im currently using `deb https://adoptopenjdk.jfrog.io/adoptopenjdk/deb/ buster main` in my sources.list, do I have to change this?

**Comment by karianna** [comment]
@aahlenst do we surface nightlies through the artifactory?

**Comment by aahlenst** [comment]
Yes, nightly builds are there. It's deb-nightly.

**Comment by aahlenst** [comment]
@TomLewis Considering the varying places where the JVM crashes, please check your memory first before doing anything else. It's more likely faulty memory than a genuine problem of the JVM. https://www.memtest86.com/ is your friend.

**Comment by TomLewis** [comment]
Im hosted with Hetzner, Will memtester do in place of memtest86? I cant see a way to boot into it from their panel.
I swapped to Oracle Java 11 for the last few days to see if that made any difference but I have been having multiple JVM crashes again, usually one after another.
if i do a full server reboot, it seems to stop it happening again.
Here are my crashes today (7 Crashes & 7 logs)
https://gist.github.com/TomLewis/976a01a23d1b615c8f161ae8db88fe1f

**Comment by aahlenst** [comment]
Memtester reveals less problems, but it's worth giving a shot.

**Comment by aahlenst** [comment]
Closing due to lack of requested feedback. We're happy to reopen the issue if feedback is provided by the OP (or anybody else) at a later date.

**Comment by TomLewis** [comment]
@aahlenst You were right!
I contacted my host and they run a memtest86 and found bad ram! 10 points! you figured it out!

**Comment by aahlenst** [comment]
😅
